src/tex/optimize.cpp

```diff
@@ -16,10 +16,14 @@ namespace btu::tex {
 
 auto optimize(Texture &&file, OptimizationSteps sets, CompressionDevice &dev) noexcept -> Result
 {
-    const auto compressed = DirectX::IsCompressed(file.get().GetMetadata().format);
-    auto res              = Result{std::move(file)};
-
-    if (compressed)
+    const auto &info           = file.get().GetMetadata();
+    const auto must_decompress = DirectX::IsCompressed(info.format);
+    // Special case - force conversion if result shouldn't have alpha to get rid of alpha bits that are added by DirectX.
+    const auto should_convert  = sets.convert || must_decompress || !DirectX::HasAlpha(sets.best_format);
+    auto res                   = Result{std::move(file)};
+
+    // We must only decompress if we want to resize or transform the image.
```

Comment:
This comment is wrong

Reply:
Changed and moved to the variable definition.